numalogic/registry/mlflow_registry.py

```diff
@@ -226,30 +226,35 @@ def transition_stage(
         """
         model_name = self.construct_key(skeys, dkeys)
         try:
-            version = int(self.get_version(model_name=model_name))
+            current_production = self.client.get_latest_versions(
+                name=model_name, stages=["Production"]
+            )
+            current_staging = self.client.get_latest_versions(name=model_name, stages=["Staging"])
+            latest = self.client.get_latest_versions(name=model_name, stages=["None"])
+
             latest_model_data = self.client.transition_model_version_stage(
                 name=model_name,
-                version=str(version),
+                version=str(latest[-1].version),
                 stage=ModelStage.PRODUCTION,
             )
-            if version - 1 > 0:
+
+            if current_production:
                 self.client.transition_model_version_stage(
                     name=model_name,
-                    version=str(version - 1),
+                    version=str(current_production[-1].version),
                     stage=ModelStage.STAGE,
                 )
-            if version - 2 > 0:
+
+            if current_staging:
                 self.client.transition_model_version_stage(
                     name=model_name,
-                    version=str(version - 2),
+                    version=str(current_staging[-1].version),
                     stage=ModelStage.ARCHIVE,
                 )
 
             # only keep "models_to_retain" number of models.
-            list_model_versions = list(self.client.search_model_versions(f"name='{model_name}'"))
-            models_to_delete = list_model_versions[: -self.models_to_retain]
-            for stale_model in models_to_delete:
-                self.delete(skeys=skeys, dkeys=dkeys, version=stale_model.version)
+            self.__delete_stale_models(skeys=skeys, dkeys=dkeys)
+
             _LOGGER.info("Successfully transitioned model to Production stage")
             return latest_model_data
         except Exception as ex:
```

Comment:
Let's only catch `RestException` here. Because anyway we are calling this under a try-except block